app/src/main/java/com/example/android/navigation/game/GameViewModel.kt

```diff
@@ -0,0 +1,134 @@
+package com.example.android.navigation.game
+
+import androidx.appcompat.app.AppCompatActivity
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+
+import com.example.android.navigation.R
+
+class GameViewModel: ViewModel(){
+
+    /**
+     * The current question in the game
+     */
+    private val  _currentQuestion = MutableLiveData<Question>()
+    val currentQuestion : LiveData<Question>
+    get() = _currentQuestion
+
+    /**
+     * The list with possible answers of the current question
+     */
+    private val _answers =  MutableLiveData<MutableList<String>>()
+    val answers : LiveData<MutableList<String>>
+    get() = _answers
+
+
+    val command = SingleLiveEvent<BaseCommand>()
+
+    /**
+     * The index of the current question.
+     */
+     var questionIndex = 0
+
+    // The first answer is the correct one.  We randomize the answers before showing the text.
+    // All questions must have four answers.  We'd want these to contain references to string
+    // resources so we could internationalize. (or better yet, not define the questions in code...)
+    private val questions: MutableList<Question> = mutableListOf(
```

Comment:
Alles dat met de Quiz en zijn vragen te maken heeft past misschien beter in een aparte "Quiz" klasse. Ik meen mij te herinneren dat ik dat gedaan had maar misschien staat dat nog op een niet-gepushte branch op mijn pc thuis. Ik check dit nog.

Reply:
Update: was idd zo, op master staat nu deze versie. Mogelijks zal merge niet perfect gaan nu

Reply:
Dit zou inderdaad beter zijn, maar ik vrees dat ik hier wat veel tijd zou door verliezen. Misschien aan de studenten laten als oefening?

Reply:
Ze hebben tegen jouw les al eens een refactor gedaan van die app, dus ze in een volgende les een niet-gerefactorde versie nog eens laten refactoren lijkt mij wat overkill.